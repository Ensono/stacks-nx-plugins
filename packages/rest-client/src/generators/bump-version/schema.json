{
    "$schema": "http://json-schema.org/schema",
    "cli": "nx",
    "$id": "BumpVersion",
    "title": "Ensono Stacks Rest Client Bump Version Generator",
    "description": "The @ensono-stacks/rest-client:bump-version generator will bump the version of any generated endpoints",
    "type": "object",
    "properties": {
        "name": {
            "type": "string",
            "description": "The endpoint you want to bump.",
            "x-prompt": "Which endpoint do you want to bump?",
            "$default": {
                "$source": "projectName"
            },
            "pattern": "^[a-z0-9_/-]+"
        },
        "endpointVersion": {
            "type": "number",
            "description": "The version you want to bump your endpoint. Omitting this value will bump latest version + 1.",
            "pattern": "^\\d+$"
        }
    },
    "required": ["name"]
}
