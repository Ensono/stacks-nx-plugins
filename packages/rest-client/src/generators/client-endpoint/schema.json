{
    "$schema": "http://json-schema.org/schema",
    "cli": "nx",
    "$id": "ClientEndpoint",
    "title": "Ensono Stacks Rest Client Client Endpoint Generator",
    "description": "The @ensono-stacks/rest-client:client-endpoint generator will add an endpoint to an existing project",
    "type": "object",
    "properties": {
        "name": {
            "type": "string",
            "description": "",
            "$default": {
                "$source": "argv",
                "index": 0
            },
            "x-prompt": "What name would you like to use?",
            "pattern": "^[a-z0-9_/-]+"
        },
        "httpClient": {
            "type": "string",
            "description": "The import path of the previously generated http-client used in the application",
            "x-prompt": "What is the import path of your previously generated http-client library? [e.g. @projectName/http-client]",
            "pattern": "^@[a-z0-9_/-]+"
        },
        "envVar": {
            "type": "string",
            "description": "The name of the API url environment variable",
            "default": "API_URL",
            "x-prompt": "What is the environment variable for the API URL?",
            "pattern": "^[A-Z_]+[A-Z0-9_]+$"
        },
        "endpointVersion": {
            "type": "number",
            "description": "The version of the endpoint",
            "default": 1,
            "x-prompt": "What is the version of your current endpoint?"
        },
        "methods": {
            "type": "array",
            "description": "List of HTTP methods to be generated. Choose from get, post, patch, put, delete, head, options.",
            "x-prompt": {
                "message": "Which methods should be generated?",
                "type": "list",
                "multiselect": true,
                "items": [
                    "get",
                    "post",
                    "patch",
                    "put",
                    "delete",
                    "head",
                    "options"
                ]
            },
            "minLength": 1
        },
        "directory": {
            "type": "string",
            "description": "Subdirectory inside libs/ where the generated library placed.",
            "x-prompt": "Where in libs/ directory do you want to generate this endpoint?",
            "alias": "dir"
        },
        "tags": {
            "type": "string",
            "description": "Add tags to the project (used for linting)",
            "alias": "t"
        }
    },
    "required": ["name", "httpClient", "envVar", "endpointVersion", "methods"]
}
