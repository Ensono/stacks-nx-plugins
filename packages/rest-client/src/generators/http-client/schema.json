{
    "$schema": "http://json-schema.org/schema",
    "cli": "nx",
    "$id": "HttpClient",
    "title": "Ensono Stacks Rest Client Http Client Generator",
    "description": "The @ensono-stacks/rest-client:http-client generator will add a rest client to an existing project",
    "type": "object",
    "properties": {
        "name": {
            "type": "string",
            "description": "Library name",
            "$default": {
                "$source": "argv",
                "index": 0
            },
            "x-prompt": "What name would you like to use for the library?",
            "pattern": "^[a-zA-Z0-9_-]+$"
        },
        "directory": {
            "type": "string",
            "description": "Subdirectory inside libs/ where the generated library placed.",
            "alias": "dir"
        },
        "importPath": {
            "type": "string",
            "description": "What import path would you like to use for the library?",
            "pattern": "^@[a-z0-9_/-]+$"
        },
        "tags": {
            "type": "string",
            "description": "Add tags to the library (used for linting).",
            "alias": "t"
        },
        "skipFormat": {
            "description": "Skip formatting files.",
            "type": "boolean",
            "default": false
        }
    },
    "required": ["name"]
}
