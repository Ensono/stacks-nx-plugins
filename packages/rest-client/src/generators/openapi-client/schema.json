{
    "$schema": "http://json-schema.org/schema",
    "$id": "OpenapiClient",
    "title": "Ensono Stacks Rest Client Openapi Client Generator",
    "description": "The @ensono-stacks/rest-client:openapi-client generator will create types, axios client, stubs and validation from an openapi spec, as a new library for an existing project",
    "type": "object",
    "properties": {
        "name": {
            "type": "string",
            "description": "Library name",
            "$default": {
                "$source": "argv",
                "index": 0
            },
            "x-prompt": "What name would you like to use for the library?",
            "pattern": "^[a-zA-Z0-9_-]+$"
        },
        "schema": {
            "type": "string",
            "description": "Provide the path to your openapi schema, relative to the root of your workspace.",
            "alias": "s",
            "x-prompt": "What is the relative path to your openapi schema?",
            "pattern": "^[a-zA-Z0-9_/.-]+$"
        },
        "zod": {
            "type": "boolean",
            "default": false,
            "description": "Zod validation for your openapi schema",
            "alias": "z",
            "x-prompt": "Do you want to generate a Zod validation for your schema?"
        },
        "tags": {
            "type": "string",
            "description": "Add tags to the project (used for linting)",
            "alias": "t"
        },
        "directory": {
            "type": "string",
            "description": "Subdirectory inside libs/ where the generated library placed."
        }
    },
    "required": ["name", "schema", "zod"]
}
