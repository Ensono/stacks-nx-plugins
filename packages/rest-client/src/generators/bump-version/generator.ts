import { normalizeOptions, copyFiles } from '@ensono-stacks/core';
import {
    formatFiles,
    names,
    readProjectConfiguration,
    Tree,
} from '@nrwl/devkit';
import { libraryGenerator } from '@nrwl/js';
import path from 'path';

import { BumpVersionGeneratorSchema } from './schema';

function findLatestVersion(
    tree: Tree,
    { directory, endpointName }: { directory?: string; endpointName: string },
): number {
    const versionsPath = readProjectConfiguration(
        tree,
        endpointName,
    ).root.replace(/v(\d+)$/g, '');

    const children = tree.children(versionsPath);

    if (children.length === 0) {
        throw new Error(
            "Could not find previous version of the endpoint. Are you sure you don't want to generate a new endpoint?",
        );
    }

    const versions = children.map(version =>
        Number.parseInt(version.replace(/^v/i, ''), 10),
    );

    return Math.max(...versions);
}

function determineNewVersion(
    latestVersion: number,
    newVersionParameter?: number | string,
): number {
    if (newVersionParameter !== undefined) {
        const newVersionNumber = Number(newVersionParameter);
        if (Number.isNaN(newVersionNumber)) {
            throw new TypeError('The endpoint version needs to be a number.');
        }
    }

    const newVersion = Number(newVersionParameter || latestVersion + 1);

    if (newVersion <= latestVersion) {
        throw new Error(
            `Cannot decrease a version. Please use --endpointVersion higher than ${latestVersion}`,
        );
    }

    return newVersion;
}

function updateVersionInCode(
    tree: Tree,
    filePaths: string[],
    latestVersion: number,
    latestVersionName: string,
    newVersion: number,
    newVersionName: string,
) {
    const latestVersionNames = names(latestVersionName);
    const newVersionNames = names(newVersionName);

    filePaths.forEach(filePath => {
        const fileContent = tree.read(filePath)?.toString();
        if (!fileContent) {
            return;
        }

        tree.write(
            filePath,
            fileContent
                .replace(
                    // eslint-disable-next-line security/detect-non-literal-regexp
                    new RegExp(latestVersionNames.className, 'g'),
                    newVersionNames.className,
                )
                .replace(
                    // eslint-disable-next-line security/detect-non-literal-regexp
                    new RegExp(latestVersionNames.name, 'g'),
                    newVersionNames.name,
                )
                .replace(
                    // eslint-disable-next-line security/detect-non-literal-regexp
                    new RegExp(`v${latestVersion}`, 'g'),
                    `v${newVersion}`,
                )
                .replace(
                    // eslint-disable-next-line security/detect-non-literal-regexp
                    new RegExp(`V${latestVersion}`, 'g'),
                    `V${newVersion}`,
                ),
        );
    });
}

export default async function bumpVersion(
    tree: Tree,
    optionsParameter: BumpVersionGeneratorSchema,
) {
    const latestVersion = findLatestVersion(tree, {
        directory: optionsParameter.directory,
        endpointName: optionsParameter.name,
    });

    if (Number.isNaN(latestVersion)) {
        throw new TypeError(
            `No version found. Please select your project with existing endpoints.`,
        );
    }

    const newVersion = determineNewVersion(
        latestVersion,
        optionsParameter.endpointVersion,
    );

    const libsEndpoint = readProjectConfiguration(
        tree,
        optionsParameter.name,
    ).root.replace(/v(\d+)$/g, '');

    const endpointRoot = libsEndpoint.replace('libs/', '');

    const latestVersionOptions = normalizeOptions(tree, {
        name: `${endpointRoot}v${latestVersion}`,
        directory: optionsParameter.directory,
        endpointName: endpointRoot,
        endpointVersion: latestVersion,
    });

    const newVersionOptions = normalizeOptions(tree, {
        ...optionsParameter,
        // include endpoint version in library name
        name: `${endpointRoot}v${newVersion}`,
        endpointName: endpointRoot,
        endpointVersion: newVersion,
    });

    // create a new library for the new version
    await libraryGenerator(tree, newVersionOptions);
    // Delete the default generated lib folder
    tree.delete(path.join(newVersionOptions.projectRoot, 'src', 'lib'));

    // copy files from latestVersion to newVersion
    copyFiles(
        tree,
        latestVersionOptions.projectRoot,
        newVersionOptions.projectRoot,
    );

    // update version numbers in the newly created files
    const filesToChange = tree
        .listChanges()
        .filter(change => change.path.startsWith(newVersionOptions.projectRoot))
        .map(change => change.path);
    updateVersionInCode(
        tree,
        filesToChange,
        latestVersion,
        latestVersionOptions.name,
        newVersion,
        newVersionOptions.name,
    );

    await formatFiles(tree);
}
