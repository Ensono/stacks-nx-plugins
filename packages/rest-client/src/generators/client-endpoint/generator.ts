import {
    createOrUpdateLocalEnv,
    verifyPluginCanBeInstalled,
    getNpmScope,
} from '@ensono-stacks/core';
import {
    formatFiles,
    generateFiles,
    getWorkspaceLayout,
    names,
    Tree,
} from '@nx/devkit';
import { determineProjectNameAndRootOptions } from '@nx/devkit/src/generators/project-name-and-root-utils';
import { libraryGenerator } from '@nx/js';
import { paramCase } from 'change-case';
import path from 'path';

import { ClientEndpointGeneratorSchema } from './schema';

const VALID_METHODS = [
    'get',
    'post',
    'patch',
    'put',
    'delete',
    'head',
    'options',
];

async function normalizeOptions(
    tree: Tree,
    options: ClientEndpointGeneratorSchema,
) {
    const name = paramCase(options.name);
    const endpointName = paramCase(`${name}/v${options.endpointVersion}`);
    let directory = path.join(name, `v${options.endpointVersion}`);

    if (options.directory) {
        directory = path.join(options.directory, directory);
    }

    if (options.projectNameAndRootFormat === 'derived') {
        const { libsDir } = getWorkspaceLayout(tree);
        directory = path.join(libsDir, directory);
    }

    const projectOptions = await determineProjectNameAndRootOptions(tree, {
        name: endpointName,
        directory,
        projectType: 'library',
        projectNameAndRootFormat: 'as-provided',
        callingGenerator: '@ensono-stacks/rest-client:client-endpoint',
    });

    return {
        ...options,
        directory,
        name: endpointName,
        projectRoot: projectOptions.projectRoot,
        projectNameAndRootFormat: 'as-provided' as const,
        importPath:
            options.importPath ?? `@${getNpmScope(tree)}/${endpointName}`,
    };
}

function addFiles(
    tree: Tree,
    options: Awaited<ReturnType<typeof normalizeOptions>>,
) {
    const templateOptions = {
        ...options,
        endpointName: names(options.name).className,
        template: '',
    };

    generateFiles(
        tree,
        path.join(__dirname, 'files'),
        options.projectRoot,
        templateOptions,
    );
}

export default async function clientEndpoint(
    tree: Tree,
    options: ClientEndpointGeneratorSchema,
) {
    verifyPluginCanBeInstalled(tree);

    const normalizedOptions = await normalizeOptions(tree, options);

    if (
        Array.isArray(normalizedOptions.methods) &&
        normalizedOptions.methods.length === 0
    ) {
        throw new Error("You haven't selected any method to generate.");
    }

    const hasInvalidMethod = normalizedOptions.methods.some(
        method => !VALID_METHODS.includes(method),
    );

    if (hasInvalidMethod) {
        throw new Error(
            `Invalid HTTP method passed to --methods. Please choose from ${VALID_METHODS.join(
                ',',
            )}`,
        );
    }

    if (Number.isNaN(Number(normalizedOptions.endpointVersion))) {
        throw new TypeError('The endpoint version needs to be a number.');
    }

    await libraryGenerator(tree, {
        ...normalizedOptions,
        bundler: 'none',
    });

    // Delete the default generated lib folder
    tree.delete(path.join(normalizedOptions.projectRoot, 'src', 'lib'));

    addFiles(tree, normalizedOptions);

    createOrUpdateLocalEnv(undefined, tree, {
        [normalizedOptions.envVar]: '',
    });

    await formatFiles(tree);
}
