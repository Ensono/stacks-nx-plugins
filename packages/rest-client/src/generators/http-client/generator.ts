import {
    addCustomTestConfig,
    NormalizedSchema as BaseNormalizedSchema,
    normalizeOptions,
} from '@ensono-stacks/core';
import {
    addDependenciesToPackageJson,
    formatFiles,
    generateFiles,
    names,
    offsetFromRoot,
    readProjectConfiguration,
    Tree,
} from '@nrwl/devkit';
import { libraryGenerator } from '@nrwl/js';
import chalk from 'chalk';
import path from 'path';

import { axiosVersion } from '../../../utils/versions';
import { HttpClientGeneratorSchema } from './schema';

type NormalizedSchema = BaseNormalizedSchema<HttpClientGeneratorSchema>;

function updateDependencies(tree: Tree) {
    return addDependenciesToPackageJson(
        tree,
        {
            axios: axiosVersion,
        },
        {},
    );
}

function addFiles(tree: Tree, options: NormalizedSchema) {
    const templateOptions = {
        ...options,
        ...names(options.name),
        offsetFromRoot: offsetFromRoot(options.projectRoot),
        template: '',
    };

    generateFiles(
        tree,
        path.join(__dirname, 'files'),
        options.projectRoot,
        templateOptions,
    );
}

export default async function generate(
    tree: Tree,
    options: HttpClientGeneratorSchema,
) {
    const normalizedOptions = normalizeOptions(tree, options);

    // Use the existing library generator
    await libraryGenerator(tree, normalizedOptions);
    // Delete the default generated lib folder
    tree.delete(path.join(normalizedOptions.projectRoot, 'src', 'lib'));

    // Generate files
    addFiles(tree, normalizedOptions);
    // Update package.json
    updateDependencies(tree);

    const project = readProjectConfiguration(
        tree,
        normalizedOptions.projectName,
    );

    const ciCoverageConfig = {
        ci: {
            collectCoverage: true,
            coverageReporters: ['text', 'html'],
            collectCoverageFrom: [
                './**/*.{js,jsx,ts,tsx}',
                './!**/.next/**',
                './!**/*.d.ts',
                './!**/*.config.*',
                './!**/_app.*',
            ],
            codeCoverage: true,
            ci: true,
        },
    };

    await addCustomTestConfig(
        tree,
        project,
        normalizedOptions.projectName,
        ciCoverageConfig,
    );

    // Format files
    if (!normalizedOptions.skipFormat) {
        await formatFiles(tree);
    }

    if (normalizedOptions.directory) {
        console.log(
            chalk.yellow`NOTE: you generated the http client inside ${normalizedOptions.directory} directory, which means that the library is now called ${normalizedOptions.projectName}`,
        );
        console.log(
            chalk.yellow`      Remember this when running nx commands like "nx test ${normalizedOptions.projectName}"`,
        );
    }
}
