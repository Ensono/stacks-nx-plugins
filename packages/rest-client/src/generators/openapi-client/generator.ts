import {
    addIgnoreEntry,
    copyFiles,
    execAsync,
    normalizeOptions,
    verifyPluginCanBeInstalled,
} from '@ensono-stacks/core';
import {
    addDependenciesToPackageJson,
    formatFiles,
    generateFiles,
    readProjectConfiguration,
    Tree,
    runTasksInSerial,
    GeneratorCallback,
} from '@nx/devkit';
import { libraryGenerator } from '@nx/js';
import { fileExists } from 'nx/src/utils/fileutils';
import path from 'path';

import { printZod } from './art';
import { OpenapiClientGeneratorSchema } from './schema';
import {
    FAKERJS_VERSION,
    MSW_VERSION,
    ORVAL_VERSION,
    ZOD_VERSION,
} from '../../../utils/versions';

export default async function generate(
    tree: Tree,
    options: OpenapiClientGeneratorSchema,
) {
    verifyPluginCanBeInstalled(tree);

    if (
        !options.schema ||
        !tree.exists(options.schema) ||
        !tree.isFile(options.schema)
    ) {
        throw new Error(
            'Provided schema does not exist in the workspace. Please check and try again.',
        );
    }

    const callbackTasks: GeneratorCallback[] = [];

    const normalizedOptions = normalizeOptions(tree, options);

    // Use the existing library generator
    await libraryGenerator(tree, normalizedOptions);

    // Delete the default generated lib folder
    tree.delete(path.join(normalizedOptions.projectRoot, 'src', 'lib'));

    const project = readProjectConfiguration(
        tree,
        normalizedOptions.projectName,
    );

    const schemaPath = path.basename(options.schema);

    // Copy schema into generated lib folder
    copyFiles(tree, `./${options.schema}`, `./${project.root}/${schemaPath}`);

    // Add prettierignore entry
    addIgnoreEntry(tree, '.prettierignore', 'openapi-client', [
        `${project.root}/src`,
    ]);

    // Add eslintignore entry
    addIgnoreEntry(tree, '.eslintignore', 'openapi-client', [
        `${project.root}/src`,
    ]);

    // Generate orval.config.js in new lib folder
    generateFiles(tree, path.join(__dirname, 'files/client'), project.root, {
        schemaName: options.name,
        schemaPath,
        template: '',
    });

    callbackTasks.push(
        addDependenciesToPackageJson(
            tree,
            {},
            {
                orval: ORVAL_VERSION,
                msw: MSW_VERSION,
                '@faker-js/faker': FAKERJS_VERSION,
            },
        ),
        () => execAsync('npm i -g orval -D', project.root) as Promise<void>,
        () =>
            execAsync(
                'orval --config ./orval.config.js',
                project.root,
            ) as Promise<void>,
    );

    if (options.zod) {
        generateFiles(tree, path.join(__dirname, 'files/zod'), project.root, {
            schemaName: options.name,
            schemaPath,
            template: '',
        });

        // Push ZOD generation to callback tasks
        callbackTasks.push(
            addDependenciesToPackageJson(
                tree,
                {
                    zod: ZOD_VERSION,
                },
                {},
            ),
            () =>
                execAsync(
                    'orval --config ./orval.zod.config.js',
                    project.root,
                ) as Promise<void>,
            () => printZod(),
        );
    }

    await formatFiles(tree);

    return runTasksInSerial(...callbackTasks);
}
