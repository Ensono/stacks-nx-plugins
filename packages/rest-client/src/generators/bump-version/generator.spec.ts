import { Tree, generateFiles } from '@nrwl/devkit';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';

import generator from './generator';
import { BumpVersionGeneratorSchema } from './schema';

jest.mock('@nrwl/devkit', () => ({
    __esModule: true,
    ...jest.requireActual('@nrwl/devkit'),
    generateFiles: jest.fn(),
}));

describe('bump-version generator', () => {
    let tree: Tree;
    const options: BumpVersionGeneratorSchema = {
        endpointPath: 'endpoints',
        endpoint: 'test',
        endpointVersion: 2,
    };

    beforeEach(async () => {
        tree = createTreeWithEmptyWorkspace();
        tree.write('endpoints/test/V1/index.ts', 'test');
    });

    it('should generate the new version of the endpoint', async () => {
        tree.write('fixtures/endpoints/test/V1/index.ts', 'test');
        await generator(tree, {
            ...options,
            endpointPath: 'fixtures/endpoints',
        });

        expect(generateFiles).toHaveBeenCalledWith(
            tree,
            'fixtures/endpoints/test/V1',
            'fixtures/endpoints/test/V2',
            {
                endpoint: 'test',
                endpointPath: 'fixtures/endpoints',
                endpointVersion: 2,
                template: '',
            },
        );
    });

    it("should throw an error if the endpoint doesn't exist", async () => {
        await expect(() =>
            generator(tree, {
                ...options,
                endpoint: 'nonexistent',
            }),
        ).rejects.toThrowError('No version could be found for endpoint');
    });

    it('should throw an error if the new version is not higher', async () => {
        await expect(() =>
            generator(tree, {
                ...options,
                endpointVersion: 1,
            }),
        ).rejects.toThrowError('Cannot bump to previous or current version');
    });
});
