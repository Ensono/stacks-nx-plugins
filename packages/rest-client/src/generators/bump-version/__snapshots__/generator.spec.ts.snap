// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bump-version generator should update any version numbers in the code 1`] = `
"import httpClient from '@next-auth/restapi/api-client';
import { AxiosRequestConfig, AxiosResponse } from 'axios';

import { TestV3, TestV3Data } from './index.types';

const API_ENDPOINT = \`\${process.env.API_URL}/test/v3\`;

export const getTestV3 = async (config: AxiosRequestConfig) =>
  httpClient.get<TestV3>(API_ENDPOINT, config);

export const deleteTestV3 = async (config: AxiosRequestConfig) =>
  httpClient.delete<TestV3>(API_ENDPOINT, config);

export const headTestV3 = async (config: AxiosRequestConfig) =>
  httpClient.head<TestV3>(API_ENDPOINT, config);

export const optionsTestV3 = async (config: AxiosRequestConfig) =>
  httpClient.options<TestV3>(API_ENDPOINT, config);

export const postTestV3 = async (
  data: TestV3Data,
  config: AxiosRequestConfig
) =>
  httpClient.post<TestV3, AxiosResponse<TestV3>, TestV3Data>(
    API_ENDPOINT,
    data,
    config
  );

export const putTestV3 = async (data: TestV3Data, config: AxiosRequestConfig) =>
  httpClient.put<TestV3, AxiosResponse<TestV3>, TestV3Data>(
    API_ENDPOINT,
    data,
    config
  );

export const patchTestV3 = async (
  data: Partial<TestV3Data>,
  config: AxiosRequestConfig
) =>
  httpClient.patch<TestV3, AxiosResponse<TestV3>, TestV3Data>(
    API_ENDPOINT,
    data,
    config
  );
"
`;

exports[`bump-version generator should update any version numbers in the code 2`] = `
"// TODO: Write generated tests

describe('TestV3 endpoint', () => {
  it('should generate the right methods', () => {
    expect(true).toBeTruthy();
  });
});
"
`;

exports[`bump-version generator should update any version numbers in the code 3`] = `
"/* eslint-disable @typescript-eslint/no-empty-interface */

export interface TestV3 {}

export interface TestV3Data {}
"
`;
